src/login.ts

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// SPDX-FileCopyrightText: 2025 Tim Beckmann <beckmann.tim@fh-swf.de>
+
+import { account } from "./appwrite";
+
+if (checkForSession()) {
+    const email = <HTMLInputElement>document.getElementById("email");
+    email.disabled = true;
+    const password = <HTMLInputElement>document.getElementById("password");
+    password.disabled = true;
+    const button = <HTMLButtonElement>document.getElementById("loginButton");
+    button.disabled = true;
+
+    const errorField = <HTMLDivElement>document.getElementById("error");
+    errorField.innerHTML=`
+        <p>Sie sind bereits angemeldet! <br />
+        Sie werden gleich weitergeleitet!</p>
+    `
+}
+
+const button = <HTMLInputElement>document.getElementById("loginButton");
+button.addEventListener('click', (event) => {
+    event.preventDefault();
+    login();
+})
+
+const form = <HTMLInputElement>document.getElementById("loginForm");
+form.addEventListener('submit', (event) => {
+    event.preventDefault();
+})
+
+function login() {
+    const email = <HTMLInputElement>document.getElementById("email");
+    const password = <HTMLInputElement>document.getElementById("password");
+    const promise = account.createEmailPasswordSession(email.value, password.value);
+
+    const errorMessage = <HTMLParagraphElement>document.getElementById("errorMessage");
+
+    promise.then(function (response) {
+        console.log(response); // Success
+        errorMessage.hidden = true;
+        document.location.href = "uebersicht.html";
+    }, function (error) {
+        console.log(error); // Failure
+        errorMessage.hidden = false;
+    })
+}
+
+function checkForSession(): boolean {
+    let sessionActive = false
+    try {
+        account.getSession("current")
+        sessionActive = true
+    } catch (e) {
+        console.error(e)
+    }
+    return sessionActive
+}
```

Comment:
Variable `sessionActive` can be optimized away.

```suggestion
function checkForSession(): boolean {
    try {
        account.getSession("current")
        return true;
    } catch (e) {
        console.error(e)
    }
    return false;
}
```